File: 11279/main.py
import sys
n = int(sys.stdin.readline()[:-1])

import heapq
heap = []
for _ in range(n):
    x = int(sys.stdin.readline()[:-1])
    if x == 0:
        if len(heap) == 0:
            print(0)
        else:
            print(heapq.heappop(heap)[1])
    else:
        heapq.heappush(heap, (-x, x))

File: 11279/main/src/main.rs
use std::collections::BinaryHeap;

fn read_int() -> i32 {
    let mut n = String::new();
    std::io::stdin().read_line(&mut n).unwrap();
    n.pop();
    n.parse::<i32>().unwrap()
}

fn main() {
    let mut heap = BinaryHeap::new();
    let n = read_int();
    
    for _ in 0..n {
        match read_int() {
            0 => match heap.pop() {
                Some(e) => println!("{}", e),
                _ => println!("0"),
            },
            x => heap.push(x),
        }
    }
}
